src/components/screens/timebank-screen.tsx

```diff
@@ -91,23 +123,41 @@ const TimebankScreen = () => {
    *
    * @param selectedDate selected date from DatePicker
    */
-  const handleDailyEntryChange = (selectedDate: DateTime) => {
-    if (selectedDate) {
-      setPersonDailyEntry(
-        dailyEntries.find((item) => DateTime.fromJSDate(item.date).day === selectedDate.day)
-      );
+  const handleDailyEntryChange = async (selectedDate: DateTime) => {
+      try {
+        const fetchedDailyEntries = await dailyEntriesApi.listDailyEntries({
+          personId: selectedEmployeeId
+        });
+        setDailyEntries(fetchedDailyEntries);
+        setPersonDailyEntry(
+          fetchedDailyEntries.find(
+            (item) =>
+              item.person === selectedEmployeeId &&
+              DateTime.fromJSDate(item.date).toISODate() === selectedDate.toISODate()
+          )
+        );
+      } catch (error) {
+        setError(`${strings.error.dailyEntriesFetch}, ${error}`);    
     }
   };
 
-  if (!personDailyEntry || !dailyEntries.length || !personTotalTime) {
-    return (
-      <Card sx={{ p: "25%", display: "flex", justifyContent: "center" }}>
-        {loading ? <CircularProgress sx={{ scale: "150%" }} /> : null}
-      </Card>
-    );
-  }
-
-  return <TimebankContent handleDailyEntryChange={handleDailyEntryChange} loading={loading} />;
+  return (
+    <div>
+      <div style={{ marginTop: "16px" }} />
+      {!personDailyEntry || !selectedEmployeeId || !dailyEntries.length || !personTotalTime ? (
+        <Card sx={{ p: "25%", display: "flex", justifyContent: "center" }}>
+          {loading ? <CircularProgress sx={{ scale: "150%" }} /> : null}
```

Comment:
Rather than using the ternary operator syntax for the conditional in this case you could just use:
{ loading && <CircularProgress sx={{ scale: "150%" }} /> }